### PR TITLE
Allow to serve WordPress via HTTP and HTTPS and don’t redirect HTTP to HTTPS.

### PR DESCRIPTION
**Pull request by ypid**
This enables TLS support by default using a self-signed certificate and redirects admin logins to it.

Related to:
- https://github.com/debops/ansible-nginx/pull/76
- https://github.com/carlalexander/debops-wordpress/issues/1

When applying this patch note that it will conflict with #36 so you have to resolve this as shown in #29.

### CODE REVIEW

**Comment by carlalexander** [comment]
When I tested this with #29, it broke nginx. The problem was that, by making `wordpress_ssl: true`, DebOps would create a HTTPS nginx config with SSL certificates using the [preconfigured mappings](https://github.com/carlalexander/debops-wordpress/blob/v0.1.0/roles/wordpress/vars/main.yml#L147-L152). But there's no certificate to load by default so nginx tries to load non-existent files and breaks.

File: roles/wordpress/vars/main.yml [hunk 1]
```diff
@@ -76,7 +82,7 @@ wordpress_backend_server:
       text/x-cross-domain-policy
       text/xml;
 
-    rewrite /wp-admin$ $scheme://$host$uri/ permanent;
+    rewrite ^/wp-admin$ $scheme://$host$uri/ permanent;
```
Comment (carlalexander) [inline roles/wordpress/vars/main.yml]: This will break for subdirectory multisite installations. That's why I left the rewrite rule as is.

Comment (ypid) [inline roles/wordpress/vars/main.yml]: Thanks. I only tested `wordpress_multisite_subdomains == True`.

**Comment by carlalexander** [comment]
I'm going to close this. We never came back to it and I'm not sure this is something we should support. In practice, we should just push for full SSL as much as possible.

**Pull request closed without merging**